components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java

```diff
@@ -114,11 +115,17 @@ public static IdentityProvider buildFederatedIdentityProvider(HttpServletRequest
                 if (diskFileItem != null) {
                     byte[] value = diskFileItem.get();
                     String key = diskFileItem.getFieldName();
-
                     if (StringUtils.equals(key, "idpUUID")) {
                         idpUUID = diskFileItem.getString();
                     }
+                    if ("meta_data_saml".equals(key)) {
 
+                        if (diskFileItem.getName() != null && diskFileItem.getName().length() > 0 && !diskFileItem.getName().toString().trim().endsWith(".xml")) {
```

Comment:
diskFileItem.getName() != null && diskFileItem.getName().length() > 0 can be simplified with StringUtils.isNotEmpty()